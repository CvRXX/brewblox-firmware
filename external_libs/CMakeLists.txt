file(GLOB nanopb_srcs nanopb/*.c)

idf_component_register(
  SRCS
    ${nanopb_srcs}
  INCLUDE_DIRS
    nanopb
    cnl/include
)
