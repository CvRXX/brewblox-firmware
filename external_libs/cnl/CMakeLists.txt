if(ESP_PLATFORM)
  idf_component_register(
    INCLUDE_DIRS
      cnl/include
    REQUIRES
      main
  )

else()
  target_include_directories(brewblox PRIVATE
    cnl/include
  )

endif()
