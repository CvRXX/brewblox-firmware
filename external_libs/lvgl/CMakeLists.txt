file(GLOB_RECURSE lvgl_srcs lvgl/src/*.c)

if(ESP_PLATFORM)
  idf_component_register(
    SRCS
      ${lvgl_srcs}
    INCLUDE_DIRS
      .
      lvgl/src
    REQUIRES
      main
  )

else()
  target_sources(brewblox PRIVATE
    ${lvgl_srcs}
  )

  target_include_directories(brewblox PRIVATE
    .
    lvgl/src
  )


endif()

add_compile_definitions(LV_LVGL_H_INCLUDE_SIMPLE)
