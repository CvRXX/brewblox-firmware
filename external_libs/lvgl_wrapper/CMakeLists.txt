# lvgl sources are included as glob to avoid LVGL component settings to be added to menuconfig
# settings are defined in lv_conf.h instead

file(GLOB_RECURSE lvgl_srcs lvgl/src/*.c)

idf_component_register(
  SRCS
    ${lvgl_srcs}
  INCLUDE_DIRS
    lvgl/src
  REQUIRES
    main
)

add_compile_definitions(LV_LVGL_H_INCLUDE_SIMPLE)
