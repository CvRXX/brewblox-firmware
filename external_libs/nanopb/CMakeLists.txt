file(GLOB nanopb_srcs nanopb/*.c)

if(ESP_PLATFORM)
  idf_component_register(
    SRCS
      ${nanopb_srcs}
    INCLUDE_DIRS
      nanopb
    REQUIRES
      main
  )

else()
  target_sources(brewblox PRIVATE
    ${nanopb_srcs}
  )

  target_include_directories(brewblox PRIVATE
    nanopb
  )

endif()
