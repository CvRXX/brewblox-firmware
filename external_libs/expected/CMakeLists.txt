if(ESP_PLATFORM)
  idf_component_register(
    INCLUDE_DIRS
      expected/include
    REQUIRES
      main
  )

else()
  target_include_directories(brewblox PRIVATE
    expected/include
  )

endif()
